core/src/test/java/com/datastax/oss/driver/internal/core/context/MockedDriverContextFactory.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright DataStax, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.datastax.oss.driver.internal.core.context;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import com.datastax.oss.driver.api.core.config.DriverConfig;
+import com.datastax.oss.driver.api.core.config.DriverConfigLoader;
+import com.datastax.oss.driver.api.core.config.DriverExecutionProfile;
+import com.datastax.oss.driver.api.core.metadata.NodeStateListener;
+import com.datastax.oss.driver.api.core.metadata.schema.SchemaChangeListener;
+import com.datastax.oss.driver.api.core.session.ProgrammaticArguments;
+import com.datastax.oss.driver.api.core.tracker.RequestTracker;
+import com.datastax.oss.driver.api.core.type.codec.TypeCodec;
+import com.datastax.oss.driver.shaded.guava.common.collect.Maps;
+import java.util.Optional;
+
+public class MockedDriverContextFactory {
+
+  public static DefaultDriverContext defaultDriverContext() {
+    return defaultDriverContext(Optional.empty());
+  }
+
+  public static DefaultDriverContext defaultDriverContext(
+      Optional<DriverExecutionProfile> profileOption) {
+
+    /* If the caller provided a profile use that, otherwise make a new one */
+    final DriverExecutionProfile profile = profileOption.orElse(mock(DriverExecutionProfile.class));
+
+    /* Setup machinery to connect the input DriverExecutionProfile to the config loader */
+    final DriverConfig driverConfig = mock(DriverConfig.class);
+    final DriverConfigLoader configLoader = mock(DriverConfigLoader.class);
+    when(configLoader.getInitialConfig()).thenReturn(driverConfig);
+    when(driverConfig.getDefaultProfile()).thenReturn(profile);
+
+    /* Setup remaining mocks */
+    final NodeStateListener nodeStateListener = mock(NodeStateListener.class);
+    final SchemaChangeListener schemaChangeListener = mock(SchemaChangeListener.class);
+    final RequestTracker requestTracker = mock(RequestTracker.class);
+    final ClassLoader classLoader = mock(ClassLoader.class);
+
+    ProgrammaticArguments args =
+        ProgrammaticArguments.builder()
+            .addTypeCodecs(new TypeCodec[0])
```

Comment:
This line can be removed.

Reply:
This is a holdover from the original impl.  Seems like there might be merit in explicitly providing something here even if the current impl of provides a sane default.  Either way, it seems like the usage here should match what we do for withLocalDatacenters() and withNodeFilters(); seems like we should rely on builder defaults for none or all three, does it not?

Reply:
Oh, nevermind, that's an "add" not a "with"... yeah, I see what you mean now.  Removing.